mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ExecutorStateManagerImpl.java

```diff
@@ -92,6 +101,10 @@ public int getTotalWorkers() {
             log.info("Archived TaskExecutor: {} removed due to: {}", notification.getKey(), notification.getCause()))
         .build();
 
+    ExecutorStateManagerImpl(String assignmentAttributesAndDefaults) {
+        this.assignmentAttributesAndDefaults = assignmentAttributesAndDefaults.isEmpty() ? ImmutableMap.of() : Splitter.on(",").withKeyValueSeparator(':').split(assignmentAttributesAndDefaults);
```

Comment:
Can we pass a Map here instead? what's the need for using primitive types here?

Reply:
agreed.